### DISCUSSION TITLE
Datastore to_tsquery Numeric Prefixing Does Not Work Well

### DISCUSSION THREAD

**Discussion by JVickery-TBS** [ckan/ckan#7916] [Ideas]
## CKAN version

2.9+ (and master branch)

## Describe the bug
Datatables view uses prefixed Full Text Search to the Datastore, which builds the `to_tsquery`. This works perfectly fine, but if you are searching with a value that begins with a number, it returns unexpected results.

### Steps to reproduce
- Load data into datastore (have some field and values that contain numbers)
- Create a datatables view
- search for a field value

example:
- Field value is `01-2019-2020-Q4`
- Search for `2020-Q4` does NOT return this row
- Searching for `0-Q4` DOES return this row

It is pretty strange behavior, I personally could not figure out how this database stuff worked.

### Expected behavior
Search results come back consistently

**Comment by EricSoroos** [comment]
I'm not seeing that straight in the database:
```
ckan=> select to_tsvector('english', '01-2019-2020-Q4')
;
            to_tsvector            
-----------------------------------
 '-2019':2 '-2020':3 '01':1 'q4':4
(1 row)

ckan=> select to_tsvector('english', '01-2019-2020-Q4') @@ to_tsquery('2020-q4')
;
 ?column? 
----------
 f
(1 row)

ckan=> select to_tsvector('english', '01-2019-2020-Q4') @@ to_tsquery('0-q4')
;
 ?column? 
----------
 f
(1 row)
ckan=> select to_tsvector('english', '01-2019-2020-Q4') @@ to_tsquery('-2020-q4')
;
 ?column? 
----------
 t
(1 row)
```

So what's happening here is that the tsvector is tokenizing to `-2020` rather than `2020`, and the `to_tsquery` values clearly don't have that until you add a `-` at the beginning.
